### DISCUSSION TITLE
CHECK constraint throws error on table creation

### DISCUSSION THREAD

**Discussion by joseph-neeraj** [simolus3/drift#1936] [Q&A] [answered]
I'm unable to create tables because of this error:

    SqliteException(1): parameters prohibited in CHECK constraints, SQL logic error (code 1)
      CREATE TABLE IF NOT EXISTS enterprise (name TEXT NOT NULL CHECK(LENGTH(name) <= ?), ..other fields);

This is the table class causing the error:

    class Enterprise extends Table {
    
      TextColumn get name =>
          text().check(name.length.isSmallerOrEqualValue(maxNameLength))();

      // ...other fields
    }

The error goes away if I remove the `check`. Could someone explain why the check isn't working ?

**Comment by simolus3** [comment, marked as answer]
Thanks for starting this discussion! I'd argue that this an oversight in drift, it will generate better in SQL in the next version (0c2ebb6c4db5944d28f21e22b74f523e452bf769).

As a workaround, you can do `check(name.length.isSmallerOrEqual(Constant(maxNameLength)))` which should generate the correct SQL without a parameter today.

**Comment by joseph-neeraj** [comment]
That worked !! Thanks 🤩
